inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -98,11 +101,19 @@ def __init__(self) -> None:
 
     def add(self, parameters: ApplicationAddSourceParameters) -> None:
         """Adds a source file and optional GPG key to be used during Ubuntu application updates."""
-        # Step 1: Add key (Optional)
+        # Step 1: Verify gpg key uri from trusted repo list 
         if parameters.gpg_key_name and parameters.gpg_key_uri:
+            try:
+               url = parameters.gpg_key_uri
+               #URL slicing to remove the last segment (filename) from the URL 
+               source = url.value[:-(len(url.value.split('/')[-1]) + 1)]
+               verify_source(source=source, dispatcher_broker=DispatcherBroker)
```

Comment:
**Suggestion:** Instead of using DispatcherBroker directly, instantiate it or ensure that it is a singleton before passing it to verify_source. Directly using a class name may lead to issues if the class is not designed to be used as a singleton. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
               dispatcher_broker = DispatcherBroker()
               verify_source(source=source, dispatcher_broker=dispatcher_broker)
```